src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2348,10 +2348,16 @@ public static string ToBase64String(ReadOnlySpan<byte> bytes, Base64FormattingOp
                 [MethodImpl(MethodImplOptions.NoInlining)]
                 static string ToBase64StringLargeInputs(ReadOnlySpan<byte> data, int outputLen)
                 {
-                    Span<byte> utf8buffer = outputLen <= 256 ? stackalloc byte[256] : new byte[outputLen];
+                    byte[]? rentedBytes = null;
+                    Span<byte> utf8buffer = outputLen <= 256 ? stackalloc byte[256] : (rentedBytes = ArrayPool<byte>.Shared.Rent(outputLen));
                     OperationStatus status = Base64.EncodeToUtf8(data, utf8buffer, out int _, out int _);
                     Debug.Assert(status == OperationStatus.Done);
-                    return Encoding.Latin1.GetString(utf8buffer);
+                    string result = Encoding.Latin1.GetString(utf8buffer);
```

Comment:
Doesn't this utf8buffer need to be sliced to the actual amount encoded?

Reply:
(And if it was passing tests without that, do we have a test hole?)

Reply:
Good catch, my understanding that we pass tests because stackalloc/Rent in most cases give us zeroed memory.

I'll make both GC.AllocateNewArray and stackalloc to always return "poisoned" data in Checked/Debug configs to catch bugs like this in a separate PR, maybe it will even help us to catch existing bugs...

Reply:
As for `EncodeToUtf8` it knows the exact output buffer size by input one - so it basically ignores size of the `utf8buffer`

Reply:
> my understanding that we pass tests because stackalloc/Rent in most cases give us zeroed memory

I don't think that can be the complete explanation here... even if utf8buffer is entirely zeros, those nulls are still going to be included in the resulting string, as GetString doesn't stop when it encounters nulls (they're not treated as termination), e.g. `Encoding.Latin1.GetString(new byte[256]).Length` == 256.

Reply:
I see you added a slice in the call to EncodeToUtf8, but the slice needs to be in the call to GetString; the Slice in the call to EncodeToUtf8 shouldn't be needed.

Reply:
@stephentoub so it turns out it wasn't properly covered, I added test cases for various lengths

Reply:
thanks.